### PR TITLE
feat: implement multi contract batch call

### PR DESCRIPTION
**Pull request by Zach-ixo**
This PR aims to reduce total RPC calls for daoContractInfo calls by batching independent queries using the multi call contract

### CODE REVIEW

File: src/utils/dao/getDaoContractInfo.ts [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+  const [admin, config, proposalModules, votingModuleAddress] = await queryMultipleContracts(daoQueries)
+  console.log('dao queries done')
+
+  const proposalModule: any = {
+    proposalConfig: config,
+    proposalModuleAddress: proposalModules[0].address,
+    votes: null,
+  }
+
+  console.log({ proposalAddress: proposalModule?.proposalModuleAddress })
```
Comment (Zach-ixo) [inline src/utils/dao/getDaoContractInfo.ts]: remove unneeded console log

File: src/utils/dao/getDaoContractInfo.ts [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+      data: {
+        proposal_creation_policy: {},
+      },
+    },
+  ]
+
+  console.log('proposal queries run')
+
+  const [proposalConfig, proposalsList, proposalCreationPolicy] = await queryMultipleContracts(proposalQueries)
+  console.log('proposal queries done')
```
Comment (Zach-ixo) [inline src/utils/dao/getDaoContractInfo.ts]: Console log

File: src/utils/multiContractCall.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import { toBase64 } from '@cosmjs/encoding'
```
Comment (Zach-ixo) [inline src/utils/multiContractCall.ts:1]: Add swimm documentation for this

File: src/utils/multiContractCall.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    if (multicallRes && multicallRes.data) {
+      const resParsed = JSON.parse(Uint8ArrayToJS(multicallRes.data))
+
+      const decoded = resParsed.return_data.map((e: any) => {
+        console.log(e)
+        return base64ToJson(e.data)
+      })
+
+      console.dir(decoded, { depth: null })
```
Comment (Zach-ixo) [inline src/utils/multiContractCall.ts]: console dir can be removed

File: src/utils/dao/getDaoContractInfo.ts [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    const total = totalValue.total
+    const stakers = listStakers.stakers
+    const config = getConfig
+
+    const cw20BaseQueries = [
+      { address: tokenContract, data: { token_info: {} } },
+      { address: tokenContract, data: { marketing_info: {} } },
+    ]
+    const [tokenInfo, marketingInfo] = await queryMultipleContracts(cw20BaseQueries)
+
```
Comment (gsSuper) [inline src/utils/dao/getDaoContractInfo.ts]: Possibly combining cw20StakeQueries, cw20BaseQueries in one queryMultipleContracts call might reduce traffic?

File: src/utils/multiContractCall.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import { toBase64 } from '@cosmjs/encoding'
+import { Uint8ArrayToJS, base64ToJson, strToArray } from './encoding'
+import { createQueryClient } from '@ixo/impactxclient-sdk'
+import { RPC_ENDPOINT } from 'lib/protocol'
+
+const contractAddress = 'ixo1rrra808ggl30g27zdmp9ecc00u7le2tn5gunv86p8aa99jrc84qqk8dttm'
```
Comment (gsSuper) [inline src/utils/multiContractCall.ts]: Is this address versatile across the different environments?

Comment (Zach-ixo) [reply src/utils/multiContractCall.ts]: I'm not sure, Michael said he'll add it to the SDK though. 

Should probably add this to a constant file for now and possibly bring in through env vars @gsSuper hey?

Comment (gsSuper) [reply src/utils/multiContractCall.ts]: Defining in env vars should work!

File: src/utils/dao/getDaoContractInfo.ts (hunk 5 above)
Comment (Zach-ixo) [reply src/utils/dao/getDaoContractInfo.ts]: Good idea 👍

**Review by alwyn-ixo — APPROVED** [review]

**Pull request merged**